diff --git a/src/misc/setrlimit.c b/src/misc/setrlimit.c
index 7a66ab29..139e1916 100644
--- a/src/misc/setrlimit.c
+++ b/src/misc/setrlimit.c
@@ -39,7 +39,8 @@ static void do_setrlimit(void *p)
 int setrlimit(int resource, const struct rlimit *rlim)
 {
 	struct ctx c = { .res = resource, .rlim = rlim, .err = -1 };
-	__synccall(do_setrlimit, &c);
+	//__synccall(do_setrlimit, &c);
+	do_setrlimit(&c);
 	if (c.err) {
 		if (c.err>0) errno = c.err;
 		return -1;
